src/main/java/models/tasks/TaskList.java

```diff
@@ -159,4 +164,41 @@ public boolean detectAnomalies(ITask newTask) {
         }
         return upcomingTasks;
     }
+
+
+    /**
+     * Returns a list of tasks in sorted order, with the amount of free time in between tasks.
+     * @param limit The time given by the user.
+     * @return A list of tasks in sorted order, with the free time between each task.
+     * @throws ParseException if date and time does not adhere to given format of dd/MM/yyyy
+     */
+    public ArrayList<String> findFreeSlots(String limit) throws ParseException {
+        ArrayList<ITask> upcomingTasks = getUpcomingTasks(limit);
+        Collections.sort(upcomingTasks, new Comparator<ITask>() {
+            @Override
+            public int compare(ITask o1, ITask o2) {
+                return o1.getDateTimeObject().compareTo(o2.getDateTimeObject());
+            }
+        });
+
+        ArrayList<String> tasksAndFreeSlots = new ArrayList<>();
+
+        Date currentDateTime = new Date(System.currentTimeMillis());
+        long diff;
+        for (int i = 0; i < upcomingTasks.size(); i++) {
+            ITask nextTaskInList = upcomingTasks.get(i);
```

Comment:
If the value `i` is not needed, consider changing the for loop to become `for (i : upcomingTasks)`